atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClient.java

```diff
@@ -23,19 +23,17 @@
 public interface CqlClient extends AutoCloseable {
 
     interface CqlQuery<R> {
-        ListenableFuture<R> execute();
+        R execute();
     }
 
-    interface CqlQueryBuilder<R> {
+    interface CqlQueryBuilder<R, T> {
```

Comment:
Yes, you are correct. It might make sense to add comments but if we know that everything is `ListenableFuture<...>` there is no need for this. I did it again to keep it as generic as possible.

Reply:
Simplified.